xr-active-switch.ts

```diff
@@ -0,0 +1,115 @@
+    scope: string | number = 2;
+
+    private components: Component[] = [];
+    private onSessionStart!: () => void;
+    private onSessionEnd!: () => void;
+
+    start() {
+        this.components = [];
+        this.collectComponents();
+
+        // Initial state setup
+        if (this.engine.xr?.session) {
+            this.applyAction(this.ifInXR);
+        } else {
+            this.applyAction(this.ifNotInXR);
+        }
+
+        // Bind event handlers
+        this.onSessionStart = this.handleSessionStart.bind(this);
+        this.onSessionEnd = this.handleSessionEnd.bind(this);
+    }
+
+    onActivate() {
+        this.engine.onXRSessionStart.add(this.onSessionStart);
+        this.engine.onXRSessionEnd.add(this.onSessionEnd);
+    }
+
+    onDeactivate() {
+        this.engine.onXRSessionStart.remove(this.onSessionStart);
+        this.engine.onXRSessionEnd.remove(this.onSessionEnd);
+    }
+
+    private collectComponents() {
+        const mode = this.resolveEnumValue(this.scope, [
+            'This Object',
+            'Children Only',
+            'This Object & Children',
+        ]);
+
+        // Handle current object
+        if (mode === 0 || mode === 2) {
+            const comps = this.object
+                .getComponents()
+                .filter((c) => c.type !== XrActiveSwitch.TypeName);
+            this.components.push(...comps);
+        }
+
+        // Handle children
+        if (mode === 1 || mode === 2) {
+            this.processChildren(this.object);
+        }
+    }
+
+    private processChildren(obj: Object3D) {
+        for (const child of obj.children) {
+            const comps = child
+                .getComponents()
+                .filter((c) => c.type !== XrActiveSwitch.TypeName);
+            this.components.push(...comps);
+            this.processChildren(child); // Recurse through all descendants
+        }
+    }
+
+    private applyAction(action: string | number) {
+        const actionMap = ['Activate', 'Deactivate', 'Toggle', 'Keep'];
+        const actionIndex = this.resolveEnumValue(action, actionMap);
+
+        for (const comp of this.components) {
+            switch (actionIndex) {
+                case 0: // Activate
+                    comp.active = true;
+                    break;
+                case 1: // Deactivate
+                    comp.active = false;
+                    break;
+                case 2: // Toggle
+                    comp.active = !comp.active;
+                    break;
+                case 3: // Keep
+                default:
+                    break;
+            }
+        }
+    }
+
+    private resolveEnumValue(value: string | number, options: string[]): number {
+        if (typeof value === 'string') {
+            return Math.max(0, options.indexOf(value));
+        }
+        return Math.max(0, Math.min(value, options.length - 1));
+    }
+
+    private handleSessionStart() {
+        this.applyAction(this.ifInXR);
+    }
```

Comment:
```suggestion
    private onSessionStart = () => {
        this.applyAction(this.ifInXR);
    }
```

Avoids .bind(this) and the extra member to store it for removal later.